### PR TITLE
feat: authentication token (ENG-789)

### PR DESCRIPTION
**Pull request by RonenMars**
## Description
Allow users token-authentication with the server.

## Linear Ticket
https://linear.app/autokitteh/issue/UI-293/configurable-token

## What type of PR is this? (check all applicable)

- [x] 💡 (feat) - A New Feature (non-breaking change which adds functionality)
- [ ] 🔄 (refactor) - Code Refactoring - A code change that neither fixes a bug nor adds a feature
- [ ] 🐞 (fix) - Bug Fix (non-breaking change which fixes an issue)
- [ ] 🏎 (perf) - Optimization
- [ ] 📄 (docs) - Documentation - Documentation only changes
- [ ] 📄 (test) - Tests - Adding missing tests or correcting existing tests
- [ ] 🎨 (style) - Styles - Changes that do not affect the meaning of the code (white spaces, formatting, missing semi-colons, etc)
- [ ] ⚙️ (ci) - Continuous Integrations - Changes to our CI configuration files and scripts (example scopes: Travis, Circle, BrowserStack, SauceLabs)
- [ ] ☑️ (chore) - Chores - Other changes that don't modify src or test files
- [ ] ↩️ (revert) - Reverts - Reverts a previous commit(s).

## Code Standards
- [x] Notify only when user attention is needed, otherwise log to console.
  - [x] Notifications: short description with context identifiers (e.g. id of the manipulated entity).
  - [x] Logs: full description with full context identifiers (e.g. deploymentId and the relevant projectId).
- [x] If you're not sure what is the proper behavior, consult with the product team.
- [x] Reduce the use of `else` by employing early returns to make code more readable and less nested.
- [x] MVC Separation: Ensure separation of concerns; views should only be manipulated by controllers (also the computing - sorting, fitlering, etc.), not by services, to maintain a clean MVC architecture.
- [x] Before implementing custom logic, check if existing functions or utilities (like lodash) offer a simpler solution.
- [x] Avoid using `!important` in CSS where possible.
- [x] Use memoization for class calculations.
- [x] Place constant strings in your code using I18n.

### CODE REVIEW

File: src/api/grpc/transport.grpc.api.ts [hunk 1]
```diff
@@ -1,7 +1,19 @@
+import { Interceptor } from "@connectrpc/connect";
 import { createConnectTransport } from "@connectrpc/connect-node";
 import { BASE_URL } from "@constants/api.constants";
+import { WorkspaceConfig } from "@utilities";
+
+export const jwtInterceptor: Interceptor = (next) => async (req) => {
```
Comment (efiShtain) [inline src/api/grpc/transport.grpc.api.ts:6]: does this have to be async ?

File: src/extension.ts [hunk 2]
```diff
@@ -43,6 +49,11 @@ export async function activate(context: ExtensionContext) {
 			return context.workspaceState.get(key, defaultValue);
 		})
 	);
+	context.subscriptions.push(
+		commands.registerCommand(vsCommands.login, async () => {
```
Comment (efiShtain) [inline src/extension.ts:53]: so you open the browser and then what ?

File: src/api/grpc/transport.grpc.api.ts (hunk 1 above)
Comment (RonenMars) [reply src/api/grpc/transport.grpc.api.ts]: Nope, removed

File: src/extension.ts (hunk 2 above)
Comment (RonenMars) [reply src/extension.ts]: Added backlink.

Also removed `async` from the function.

File: src/extension.ts [hunk 3]
```diff
@@ -43,6 +49,24 @@ export async function activate(context: ExtensionContext) {
 			return context.workspaceState.get(key, defaultValue);
 		})
 	);
+	context.subscriptions.push(
+		commands.registerCommand(vsCommands.login, () => {
+			env.openExternal(Uri.parse(`${BASE_URL}/login`));
+		})
+	);
+	const handleUri = (uri: Uri) => {
```
Comment (RonenMars) [inline src/extension.ts:57]: Remove and add a command palette to change the token.

**Pull request merged**